R/cellbrowser.R

```diff
@@ -0,0 +1,499 @@
+#' @param matrix.slot matrix to use, default is 'counts'
+#' @param use.mtx export the matrix in .mtx.gz format. Default is False,
+#'        unless the matrix is bigger than R's maximum matrix size.
+#' @param cluster.field name of the metadata field containing cell cluster
+#' @param cb.dir path to directory where to create UCSC cellbrowser static
+#' website content root, e.g. an index.html, .json files, etc. These files
+#' can be copied to any webserver. If this is specified, the cellbrowser
+#' package has to be accessible from R via reticulate.
+#' @param meta.fields vector of meta fields to export, default is all.
+#' @param meta.fields.names vector meta field names to show in UI. Must have
+#'        same length as meta.fields. Default is meta.fields.
+#' @param skip.markers whether to skip exporting markers
+#' @param skip.expr.matrix whether to skip exporting expression matrix
+#' @param skip.metadata whether to skip exporting metadata
+#' @param skip.reductions whether to skip exporting reductions
+#' @param port on which port to run UCSC cellbrowser webserver after export
+#' @param ... specifies the metadata fields to export. To supply a field and its
+#' human readable name, pass name as \code{field="name"} parameter.
+#'
+#' @return This function exports Seurat object as a set of tsv files
+#' to \code{dir} directory, copying the \code{markers.file} if it is
+#' passed. It also creates the default \code{cellbrowser.conf} in the
+#' directory. This directory could be read by \code{cbBuild} to
+#' create a static website viewer for the dataset. If \code{cb.dir}
+#' parameter is passed, the function runs \code{cbBuild} (if it is
+#' installed) to create this static website in \code{cb.dir} directory.
+#' If \code{port} parameter is passed, it also runs the webserver for
+#' that directory and opens a browser.
+#'
+#' @author Maximilian Haeussler, Nikolay Markov
+#'
+#' @importFrom tools file_ext
+#' @importFrom utils browseURL
+#' @importFrom reticulate py_module_available import
+#' @importFrom Seurat Project Idents GetAssayData Embeddings FetchData
+#'
+#' @export
+#'
+#' @name CellBrowser
+#' @rdname CellBrowser
+#'
+#' @importFrom methods slot
+#' @importFrom utils packageVersion
+#' @importFrom reticulate py_module_available import
+#'
+#' @examples
+#' \dontrun{
+#' ExportToCellbrowser(pbmc_small, dataset.name = "PBMC", dir = "out")
+#' }
+#'
+ExportToCellbrowser <- function(
+  object,
+  dir,
+  dataset.name = Project(object = object),
+  reductions = NULL,
+  markers.file = NULL,
+  cluster.field = NULL,
+  cb.dir = NULL,
+  port = NULL,
+  use.mtx = FALSE,
+  meta.fields = NULL,
+  meta.fields.names = NULL,
+  matrix.slot = "counts",
+  markers.n = 100,
+  skip.markers = FALSE,
+  skip.expr.matrix = FALSE,
+  skip.metadata = FALSE,
+  skip.reductions = FALSE
+) {
+  if (!requireNamespace("Seurat", quietly = TRUE)) {
+    stop("This script requires that Seurat (V2 or V3) is installed")
+  }
```

Comment:
At least in SeuratWrappers, Seurat is assumed to be installed; is this check necessary?

Reply:
Not necessary here, but could I leave it in? It has happened to me and colleagues a few times that we got very strange error messages in pipelines and this just confirms something which should be the case anyways and for the unlikely case that it happens, has a nice error message.

Reply:
Were you getting error messages when using this function from within SeuratWrappers? If so, then there's a bug in SeuratWrappers somewhere as this package depends on Seurat

Reply:
We use this function in a little script where the incoming data is a Seurat object. It uses whatever R and Seurat the user has installed and the function is run to convert the data. So it's just another check, it's rare, but some users have different R environments and don't fully know which one they're running the script in, so I'd rather leave the check in.